pennylane/registers.py

```diff
@@ -0,0 +1,70 @@
+"""
+This module contains the :func:`registers` function to build the registers for a given dictionary of registers
+"""
+
+from .wires import Wires
+
+
+def registers(register_dict, _start_wire_index=0):
```

Comment:
The registers function is recursive and uses the _start_wire_index for the starting index of the nested registers.

For example in this case: `{"alice": 3, "bob": {"nest1": {"nest2": {"nest3": 1}}}, "cleo": 3}`, bob and the nested elements of bob will all start at the same index (2), without `_start_wire_index` I'm not sure how to get the same functionality

Reply:
If this is a user-facing function, we could have a wrapper to stop exposing `_start_wire_index` to them? Otherwise, we would name a better/cleaner name to mention it in the UI.